src/initialization/MOM_state_initialization.F90

```diff
@@ -506,6 +507,11 @@ subroutine MOM_initialize_state(u, v, h, tv, Time, G, GV, US, PF, dirs, &
   ! The thicknesses in halo points might be needed to initialize the velocities.
   if (new_sim) call pass_var(h, G%Domain)
 
```

Comment:
The recent refactoring of the OBC initialization code attempts to collect all of the OBC segment initialization into the separate routine `MOM_initialize_OBCs()` which appears starting at about line 650 of this file.  Please move this added block into `MOM_initialize_OBCs()`.